tools/common/Assembly.cs

```diff
@@ -464,4 +477,75 @@ public void CopySatellitesToDirectory (string directory)
 			}
 		}
 	}
+
+	public class AssemblyCollection : IEnumerable<Assembly>
+	{
+		Dictionary<string, Assembly> HashedAssemblies = new Dictionary<string, Assembly> (StringComparer.OrdinalIgnoreCase);
+
+		public void Add (Assembly assembly)
+		{
+			Assembly other;
+			if (HashedAssemblies.TryGetValue (assembly.Identity, out other))
+				throw ErrorHelper.CreateError (2018, "The assembly '{0}' is referenced from two different locations: '{1}' and '{2}'.", assembly.Identity, other.FullPath, assembly.FullPath);
+			HashedAssemblies.Add (assembly.Identity, assembly);
+		}
+
+		public void AddRange (AssemblyCollection assemblies)
+		{
+			if (Count == 0) {
+				HashedAssemblies = new Dictionary<string, Assembly> (assemblies.HashedAssemblies);
```

Comment:
I thought I had looked at the Dictionary source code, and found that when creating a new instance from a dictionary, the dictionary would be smart enough to not add the elements one-by-one.

I must have dreamt, because I just looked again, and the dictionary constructor adds the elements one-by-one.